### PR TITLE
Add back machine driver for skolas proto

### PR DESCRIPTION
**Pull request by cujomalainey**
~Also a IPC4 build error fix~

### CODE REVIEW

File: sound/soc/sof/ipc4-topology.c [hunk 1]
```diff
@@ -882,7 +882,7 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 	/* allocate memory for base config extension if needed */
 	if (process->init_config == SOF_IPC4_MODULE_INIT_CONFIG_TYPE_BASE_CFG_WITH_EXT) {
 		struct sof_ipc4_base_module_cfg_ext *base_cfg_ext;
-		u32 ext_size;
+		u32 ext_size = 0;
 
 		ext_size += struct_size(base_cfg_ext, pin_formats,
```
Comment (bardliao) [inline sound/soc/sof/ipc4-topology.c]: Or just use `=` here. Like `ext_size = struct_size(base_cfg_ext, pin_formats,`

Comment (ujfalusi) [reply sound/soc/sof/ipc4-topology.c]: or `u32 ext_size = struct_size(base_cfg_ext, pin_formats,`, but the suggestion from @bardliao might look better.

@cujomalainey, can you separate this patch as a separate PR?

File: sound/soc/intel/boards/sof_rt5682.c [hunk 2]
```diff
@@ -1127,6 +1127,17 @@ static const struct platform_device_id board_ids[] = {
...
+		.name = "rpl_mx98360_rt5682",
+		.driver_data = (kernel_ulong_t)(SOF_RT5682_MCLK_EN |
+					SOF_RT5682_SSP_CODEC(0) |
+					SOF_SPEAKER_AMP_PRESENT |
+					SOF_MAX98360A_SPEAKER_AMP_PRESENT |
+					SOF_RT5682_SSP_AMP(1) |
+					SOF_RT5682_NUM_HDMIDEV(4) |
+					SOF_BT_OFFLOAD_SSP(2) |
+					SOF_SSP_BT_OFFLOAD_PRESENT),
+	},
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_rt5682.c]: nit-pick: can you move this between adl and mtl, we mostly use soc generations to sort the driver_data (except for jsl which seems to be the exception to the rule).

File: sound/soc/sof/ipc4-topology.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/ipc4-topology.c]: I also got it today from 0day with clang.

````
>> sound/soc/sof/ipc4-topology.c:854:3: warning: variable 'ext_size' is uninitialized when used here [-Wuninitialized]
                   ext_size += struct_size(base_cfg_ext, pin_formats,
                   ^~~~~~~~
   sound/soc/sof/ipc4-topology.c:852:15: note: initialize the variable 'ext_size' to silence this warning
                   u32 ext_size;
                               ^
                                = 0
   1 warning generated.
````

This should be a separate PR, that's possibly a fixup!

File: sound/soc/intel/boards/sof_rt5682.c (hunk 2 above)
Comment (cujomalainey) [reply sound/soc/intel/boards/sof_rt5682.c]: Sure, was trying to go alphabetical but if that is the method I can switch

File: sound/soc/sof/ipc4-topology.c (hunk 1 above)
Comment (cujomalainey) [reply sound/soc/sof/ipc4-topology.c]: #4214

**Review by plbossart — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Pull request merged**